src/pt/mediocretoons/src/eu/kanade/tachiyomi/extension/pt/mediocretoons/MediocreToons.kt

```diff
@@ -1,19 +1,285 @@
 package eu.kanade.tachiyomi.extension.pt.mediocretoons
 
-import eu.kanade.tachiyomi.multisrc.greenshit.GreenShit
+import eu.kanade.tachiyomi.network.GET
 import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import keiyoushi.utils.parseAs
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
 
-class MediocreToons : GreenShit(
-    "Mediocre Toons",
-    "https://mediocretoons.com",
-    "pt-BR",
-    scanId = 2,
-) {
-    override val targetAudience = TargetAudience.Shoujo
+class MediocreToons : HttpSource() {
 
-    override val contentOrigin = ContentOrigin.Mobile
+    override val name = "Mediocre Toons"
 
-    override val client = super.client.newBuilder()
+    override val baseUrl = "https://mediocretoons.com"
+
+    override val lang = "pt-BR"
+
+    override val supportsLatest = true
+
+    private val apiUrl = "https://api.mediocretoons.com"
+
+    private val scanId: Long = 2
+
+    override val client = network.cloudflareClient.newBuilder()
+        .addInterceptor(::imageLocation)
         .rateLimit(2)
         .build()
+
+    override fun headersBuilder() = super.headersBuilder()
+        .set("scan-id", scanId.toString())
+        .set("x-app-key", "toons-mediocre-app")
+
+    // ============================== Popular ================================
+    override fun popularMangaRequest(page: Int) =
+        GET("$apiUrl/obras?ordenarPor=views_hoje&limite=20", headers)
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val dto = response.parseAs<MediocreListDto<List<MediocreMangaDto>>>()
+        val mangas = dto.data.map { it.toSManga() }
+        val hasNext = dto.pagination?.hasNextPage ?: false
+        return MangasPage(mangas, hasNextPage = hasNext)
```

Comment:
If there are no next pages in popular manga
```suggestion
        return MangasPage(mangas, hasNextPage = false)
```